nearcore/src/config_validate.rs

```diff
@@ -75,107 +83,116 @@ impl<'a> ConfigValidator<'a> {
+        }
+    }
+
+    fn validate_state_dumper_config(&mut self, dump_config: &DumpConfig) {
+        if let Some(restart_dump_for_shards) = &dump_config.restart_dump_for_shards {
+            let unique_values: HashSet<_> = restart_dump_for_shards.iter().collect();
+            if unique_values.len() != restart_dump_for_shards.len() {
+                let error_message = format!(
+                    "'config.state_sync.dump.restart_dump_for_shards' contains duplicate values."
+                );
+                self.validation_errors.push_config_semantics_error(error_message);
+            }
+        }
+
+        match &dump_config.location {
+            ExternalStorageLocation::S3 { bucket, region } => {
+                if bucket.is_empty() || region.is_empty() {
+                    let error_message = format!(
+                        "'config.state_sync.dump.location.S3.bucket' and 'config.state_sync.dump.location.S3.region' need to be specified when 'config.state_sync.dump.location.S3' is present."
+                    );
+                    self.validation_errors.push_config_semantics_error(error_message);
+                }
+            }
+            ExternalStorageLocation::Filesystem { root_dir } => {
+                if root_dir.as_path() == Path::new("") {
+                    let error_message = format!(
+                        "'config.state_sync.dump.location.Filesystem.root_dir' needs to be specified when 'config.state_sync.dump.location.Filesystem' is present."
+                    );
+                    self.validation_errors.push_config_semantics_error(error_message);
                 }
+            }
+            ExternalStorageLocation::GCS { bucket } => {
+                if bucket.is_empty() {
+                    let error_message = format!(
+                        "'config.state_sync.dump.location.GCS.bucket' needs to be specified when 'config.state_sync.dump.location.GCS' is present."
+                    );
+                    self.validation_errors.push_config_semantics_error(error_message);
+                }
+            }
+        }
 
-                match &dump_config.location {
+        if let Some(credentials_file) = &dump_config.credentials_file {
+            if !credentials_file.exists() || !credentials_file.is_file() {
+                let error_message = format!(
+                    "'config.state_sync.dump.credentials_file' is provided but the specified file does not exist or is not a file."
+                );
+                self.validation_errors.push_config_semantics_error(error_message);
+            }
+        }
+
+        if let Some(parts_compression_lvl) = dump_config.parts_compression_lvl {
+            if parts_compression_lvl < -22 || parts_compression_lvl > 22 {
+                let error_message = format!(
+                    "'config.state_sync.dump.parts_compression_lvl' should be an integer between -22 and 22."
```

Comment:
nit: to be consistent with other errors (looking at tx_routing_height_horizon checks) and make error slightly more descriptive, consider including the value of compression lvl we got.